scheduler.rst

```diff
@@ -0,0 +1,328 @@
+While one could resort to issuing commands and scheduling them with cron jobs, this approach involves external tools and additional configuration.
+
+The Scheduler component emerges as a solution, allowing you to retain control, configuration, and maintenance of task scheduling within our PHP application.
+
+At its core, the principle is straightforward: a task, considered as a message needs to be managed by a service, and this cycle must be repeated.
+Does this sound familiar? Think :doc:`Symfony Messenger docs </components/messenger>`.
+
+But while the system of Messenger proves very useful in various scenarios, there are instances where its capabilities
+fall short, particularly when dealing with repetitive tasks at regular intervals.
+
+Let's dive into a practical example within the context of a sales company.
+
+Imagine the company's goal is to send diverse sales reports to customers based on the specific reports each customer chooses to receive.
+In constructing the schedule for this scenario, the following steps are taken:
+
+#. Iterate over reports stored in the database and create a recurring task for each report, considering its unique properties. This task, however, should not be generated during holiday periods.
+
+#. Furthermore, you encounter another critical task that needs scheduling: the periodic cleanup of outdated files that are no longer relevant.
+
+On the basis of a case study in the context of a full stack Symfony application, let's dive in and explore how you can set up your system.
+
+Symfony Scheduler basics
+------------------------
+
+Differences and parallels between Messenger and Scheduler
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The primary goal is to generate and process reports generation while also handling the removal of outdated reports at specified intervals.
+
+As mentioned, this component, even if it's an independent component, it draws its foundation and inspiration from the Messenger component.
+
+On one hand, it adopts well-established concepts from Messenger (such as message, handler, bus, transport, etc.).
+For example, the task of creating a report is considered as a message that will be directed, and processed by the corresponding handler.
+
+However, unlike Messenger, the messages will not be dispatched in the first instance. Instead, the aim is to create them based on a predefined frequency.
+
+This is where the specific transport in Scheduler, known as the :class:`Symfony\\Component\\Scheduler\\Messenger\\SchedulerTransport`, plays a crucial role.
+The transport autonomously generates directly various messages according to the assigned frequencies.
+
+From (Messenger cycle):
+
+.. image:: /_images/components/messenger/basic_cycle.png
+    :alt: Symfony Messenger basic cycle
+
+To (Scheduler cycle):
+
+.. image:: /_images/components/scheduler/scheduler_cycle.png
+    :alt: Symfony Scheduler basic cycle
```

Comment:
I like these diagrams. But it may be too much for this early in the docs.

Reply:
Shouldn't we distinguish between the Messenger cycle and that of the Scheduler from the outset?